compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -285,7 +288,6 @@ impl<'context> Elaborator<'context> {
 
     fn elaborate_function(&mut self, mut function: NoirFunction, id: FuncId) {
         self.current_function = Some(id);
-        self.resolve_where_clause(&mut function.def.where_clause);
```

Comment:
A few things within elaborate_function have been split into define_function_meta.

Generally items to do with the function signature should be handled beforehand